views/AddCurrency/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { useDispatch } from 'react-redux'
+import React, {useState, useEffect, FC} from 'react';
+import { Text, TextInput, TouchableOpacity, View, SafeAreaView } from 'react-native';
+import {addCurrencyBySymbol} from '../../redux/actions/currenciesActions';
+import colors from '../../assets/stylesRoot/colors';
+import { useNavigation } from '@react-navigation/native';
+import { Routes } from '../../navigation/routes';
+import styles from './styles';
+
+const AddCurrency= (): JSX.Element => {
+  const dispatch = useDispatch()
+  const navigation: any = useNavigation();
+  const routes = useNavigation().getState()?.routes
+  const prevRoute = routes[routes.length - 2]
+
+  const [inputValue, setInputValue] = useState<string>('')
+  const [isFocused, setIsFocused] = useState<boolean>(false)
+  const [isButtonActive, setIsButtonActive] = useState<boolean>(false)
+  const [error, setError ] = useState<string>('')
+
+  const handleFocus = () => setIsFocused(true) 
+  const handleBlur = () => setIsFocused(false) 
```

Comment:
[suggestion] Since no extra work is being done on these callbacks, you might as well use bind() and use them in the onPress property